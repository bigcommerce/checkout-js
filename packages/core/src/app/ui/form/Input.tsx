import classNames from 'classnames';
import { includes } from 'lodash';
import React, { forwardRef, InputHTMLAttributes, Ref } from 'react';

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
    testId?: string;
    useFloatingLabel?: boolean;
}

const Input = forwardRef(
    (
        { className, testId, placeholder, name, useFloatingLabel, ...rest }: InputProps,
        ref: Ref<HTMLInputElement>,
    ) => {
        const notUseFloatingLabelFieldNames = ['orderComment', 'redeemableCode'];

        return (
            <input
                {...{
                    ...rest,
                    className: classNames(className, {
                        'floating-input':
                            useFloatingLabel && !includes(notUseFloatingLabelFieldNames, name),
                    }),
                    name,
                }}
                data-test={testId}
                placeholder={useFloatingLabel ? ' ' : placeholder}
                ref={ref}
            />
        );
    },
);

export default Input;
