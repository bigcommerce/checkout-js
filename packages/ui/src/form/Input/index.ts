export { default as Input, InputProps } from './Input';
