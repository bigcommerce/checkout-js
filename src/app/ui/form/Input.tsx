import React, { forwardRef, InputHTMLAttributes, Ref } from 'react';

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
    testId?: string;
}

const Input = forwardRef((
    { testId, disabled, ...rest }: InputProps,
    ref: Ref<HTMLInputElement>
) => (
    <input
        { ...rest }
        data-test={ testId }
        ref={ ref }
        disabled={ disabled ? disabled : false }
    />
));

export default Input;
