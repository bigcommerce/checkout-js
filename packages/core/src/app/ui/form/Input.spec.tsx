import { shallow } from 'enzyme';
import React from 'react';

import Input from './Input';

describe('Input', () => {
    it('matches snapshot', () => {
        expect(shallow(<Input name="foobar" />)).toMatchSnapshot();
    });

    it('renders component with test ID', () => {
        expect(shallow(<Input name="foobar" testId="test" />).prop('data-test')).toBe('test');
    });

    it('listens to DOM events', () => {
        const handleChange = jest.fn();
        const component = shallow(<Input name="foobar" onChange={handleChange} />);

        component.simulate('change');

        expect(handleChange).toHaveBeenCalled();
    });
});
