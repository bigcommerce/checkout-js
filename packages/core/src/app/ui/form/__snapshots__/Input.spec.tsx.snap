// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<input
  name="foobar"
/>
`;
